Source/WebCore/svg/SVGPathElement.cpp

```diff
@@ -94,6 +97,7 @@ void SVGPathElement::attributeChanged(const QualifiedName& name, const AtomStrin
 void SVGPathElement::clearCache()
 {
     pathSegListCache().clear();
+    pathSegListCacheSize = 0;
```

Comment:
The HashMap and the data size can be members of the new class and this function should call 
```
PathSegListCache::singleton().clear();
```